### DISCUSSION TITLE
Katex and remarkMath are not working based on the example

### DISCUSSION THREAD

**Discussion by ShurikAg** [remarkjs/remark#1376] [Q&A] [answered]
Hi all, 

I am using the example provided: https://github.com/remarkjs/react-markdown?tab=readme-ov-file#use-remark-and-rehype-plugins-math

So, my code looks something like this:

```
import Markdown from 'react-markdown'
import rehypeKatex from 'rehype-katex'
import remarkMath from 'remark-math'
import 'katex/dist/katex.min.css'

...
<Markdown remarkPlugins={[remarkMath]} rehypePlugins={[rehypeKatex]}>{message.content}</Markdown>
```

The markdown is getting parsed properly, but it seems like the formulas are not getting picked up.
<img width="949" alt="image" src="https://github.com/user-attachments/assets/5768930e-48ba-4b4a-98bc-68556a454b49">

I am not 100% sure how to debug it. There are no errors or anything.

**Comment by ChristianMurphy** [comment]
@ShurikAg it look like you want mhchem?
That's requires sn extra extension https://mhchem.github.io/MathJax-mhchem/ or https://github.com/KaTeX/KaTeX/blob/main/contrib/mhchem/README.md

**Reply by ShurikAg** [reply]
Thank you, I am struggling to understand how to add the extension. Do I need the entire `MathJax-mhchem` package or is there a lightweight one available?
 Honestly, I am not finding any of these.

My apologies, this is a first time I am trying to use these libs, so still trying to understand the structure.

**Reply by ChristianMurphy** [reply]
No worries, when you have a chance could you try the solution here: https://github.com/remarkjs/remark-math/issues/81#issuecomment-1434803831 ?

**Reply by ShurikAg** [reply]
Hmm... this is what I have right now:

```
import Markdown from 'react-markdown';
import remarkParse from 'remark-parse';
import remarkMath from 'remark-math';
import remarkRehype from 'remark-rehype';
import rehypeKatex from 'rehype-katex';
import rehypeStringify from 'rehype-stringify';
import 'katex/dist/contrib/mhchem';
import 'katex/dist/katex.min.css';

...
<Markdown remarkPlugins={[remarkParse, remarkMath, remarkRehype]}
                    rehypePlugins={[rehypeKatex, rehypeStringify]}>
            {message.content}
</Markdown>
```

And nothing changed

**Reply by ChristianMurphy** [reply]
@ShurikAg could you also share the markdown text you are trying to render?

**Reply by ShurikAg** [reply]
This is the raw content:

```
The reduction of tantalum pentoxide (Ta2O5) using magnesium (Mg) and carbon (C) involves thermochemical processes that can be analyzed in terms of thermal effects, such as enthalpy changes and reaction temperatures.

1. **Reduction of Ta2O5 with Magnesium (Mg)**:
- The reaction can be represented as:
\[
\text{Ta}_2\text{O}_5 + 5\text{Mg} \rightarrow 2\text{Ta} + 5\text{MgO}
\]
- This reaction is typically exothermic, meaning it releases heat. The reduction occurs at elevated temperatures, often above 800°C, where magnesium can effectively reduce Ta2O5 to tantalum metal.
- The enthalpy change (ΔH) for this reaction is negative, indicating that the products (tantalum and magnesium oxide) are at a lower energy state than the reactants.

2. **Reduction of Ta2O5 with Carbon (C)**:
- The reaction can be represented as:
\[
\text{Ta}_2\text{O}_5 + 5\text{C} \rightarrow 2
```

This is a raw output from ChatGPT

**Reply by ShurikAg** [reply]
Looks like ChatGPT themselves are using Katex :) 

<img width="1001" alt="image" src="https://github.com/user-attachments/assets/3a7b13fc-39bb-4c49-8661-f95be23cfdc4">

**Reply by ChristianMurphy** [reply, marked as answer]
See https://github.com/remarkjs/react-markdown/issues/785

Ask ChatGPT to use `$$` as the latex markers.
